streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -98,4 +116,47 @@ public String toString() {
         }
         return sb.toString();
     }
+
+    protected class NodeMetrics  {
+        final StreamsMetrics metrics;
+        final String metricGrpName;
+        final Map<String, String> metricTags;
+
+        final Sensor nodeProcessTimeSensor;
+        final Sensor nodePunctuateTimeSensor;
+        final Sensor contextForwardSensor;
+        final Sensor nodeTaskCreationSensor;
+        final Sensor nodeTaskDestructionSensor;
+
+
+        public NodeMetrics(StreamsMetrics metrics, String name, String sensorNamePrefix) {
+            String scope = "node";
+            this.metrics = metrics;
+            this.metricGrpName = "node-metrics-" + name;
+            this.metricTags = new LinkedHashMap<>();
+            this.metricTags.put("node-id", "-" + name);
+
+            // these are all latency metrics
+            this.nodeProcessTimeSensor = metrics.addLatencySensor(scope, name, "process", Sensor.RecordLevel.SENSOR_DEBUG);
+            this.nodePunctuateTimeSensor = metrics.addLatencySensor(scope, name, "punctuate", Sensor.RecordLevel.SENSOR_DEBUG);
+
+            this.contextForwardSensor = metrics.sensor(sensorNamePrefix + "node-forward-time" + name, Sensor.RecordLevel.SENSOR_DEBUG);
```

Comment:
@guozhangwang @aartigupta Unfortunately 2 different tasks on the same thread can register the same two metrics and we'll get an exception in the thread library (current test fails because of this). So the prefix is not sufficient just for the sensor, but is also needed for the metric.